checkov/terraform/graph_builder/foreach/abstract_handler.py

```diff
@@ -212,67 +149,101 @@ def _update_attributes(self, attrs: dict[str, Any], key_to_val_changes: dict[str
         return foreach_attributes
 
     @staticmethod
-    def _build_key_to_val_changes(new_val: str, new_key: str):
-        return {
-            EACH_VALUE: new_val,
-            EACH_KEY: new_key
-        }
+    def _update_block_name_and_id(block: TerraformBlock, idx: int | str) -> str:
+        # Note it is important to use `\"` inside the string,
+        # as the string `["` is the separator for `foreach` in terraform.
+        # In `count` it is just `[`
+        idx_with_separator = f'\"{idx}\"' if isinstance(idx, str) else f'{idx}'
+        new_block_id = f"{block.id}[{idx_with_separator}]"
+        new_block_name = f"{block.name}[{idx_with_separator}]"
 
-    def _create_new_resource(
-            self,
-            main_resource: TerraformBlock,
-            new_value: int | str,
-            resource_idx: int,
-            foreach_idx: int,
-            new_key: Optional[str] = None,
-    ):
-        new_resource = deepcopy(main_resource)
-        block_type, block_name = new_resource.name.split('.')
-        if main_resource.attributes.get(COUNT_STRING):
-            key_to_val_changes = {COUNT_KEY: new_value}
-        else:
-            key_to_val_changes = self._build_key_to_val_changes(new_value, new_key)
-        config_attrs = new_resource.config.get(block_type, {}).get(block_name, {})
-        self._pop_foreach_attrs(new_resource.attributes)
-        self._pop_foreach_attrs(config_attrs)
-        self._update_attributes(new_resource.attributes, key_to_val_changes)
-        foreach_attrs = self._update_attributes(config_attrs, key_to_val_changes)
-        new_resource.foreach_attrs = foreach_attrs
+        if block.block_type == BlockType.MODULE:
```

Comment:
looks like this is a leftover from before the refactor, should be moved to `ForeachModuleHandler`